src/vcBPA.cpp

```diff
@@ -924,86 +954,97 @@ vdkError vcBPA_ConvertReadPoints(vdkConvertCustomItem *pConvertInput, vdkPointBu
   pBuffer->pointCount = 0;
 
   vcBPAConvertItem *pData = (vcBPAConvertItem *)pConvertInput->pData;
-  bool getNextGrid = true;
-
-  uint32_t displacementOffset = 0;
-  vdkError error = vE_Failure;
 
   static int gridCount = 0;
+  if (gridCount == 0)
+  {
+    do
+    {
+      if (udSafeDeque_PopFront(pData->pConvertItemData, &pData->activeItem) == udR_Success)
+        break;
+      udSleep(250);
+    } while (pData->running);
+  }
+
   if (pData->activeItem.pointIndex == 0)
+  {
     ++gridCount;
+    if (pData->activeItem.pGrid == nullptr)
+      goto epilogue;
+  }
 
-  if (pData->activeItem.pointIndex == 0 && pData->activeItem.pGrid == nullptr)
-    goto epilogue;
-
-  error = vdkAttributeSet_GetOffsetOfNamedAttribute(&pBuffer->attributes, "udDisplacement", &displacementOffset);
-  if (error != vE_Success)
-    return error;
-
-  for (size_t i = pData->activeItem.pointIndex; i < pData->activeItem.pGrid->vertices.length; ++i)
+  while (pBuffer->pointCount < pBuffer->pointsAllocated)
```

Comment:
Unless I'm mistaken, this is changing the callback to process multiple grids per callback instead of just one? I highly recommend reverting this change, this is adding too much complexity to the PR. I don't think it's a necessary optimization at this point either, but for now we need to focus on the octree changes and nothing else.